content/blog/elm-at-noredink/index.md

```diff
@@ -0,0 +1,445 @@
+---
+title: "Elm at NoRedInk"
+date: "2021-02-15T12:00:00.000Z"
+description: How to organize 500K lines of Elm.
+---
+
+At [NoRedInk](https://www.noredink.com) we have one of the largest Elm apps
+in the world. It serves millions of teachers and students every day and our
+frontend is exclusively written in Elm. In this post, we will explore the
+structure of our application and the patterns that we use to stay sane.
+
+One of the most common questions I get about Elm is "But does it scale? And
+if so, how". Well, I hope that this post can become a comprehensive answer
+to that question.
+
+## Some stats first
+
+We have a glorious monorepo which contains all the services that power our
+infrastructure. Our main application (aka the `monolith`) is written in
+Rails and has a `ui` folder with all of our Elm code.
+
+```bash
+$ find monolith/ui/src -name "*.elm" | xargs cat | wc -l
+275082
+
+$ find monolith/ui/tests -name "*.elm" | xargs cat | wc -l
+209107
+```
```

Comment:
💭  I was surprised to see that `cloc` throws a ratio of 0.95 lines of test/implementation code, higher than the 76% we get with `wc -l`!

the output is more verbose and some readers may not be familiar with the program though so not suggesting we change anything.